### PR TITLE
semver module

### PR DESCRIPTION
**Pull request by rubiojr**
Fixes https://github.com/risor-io/risor/issues/260

### CODE REVIEW

**Comment by rubiojr** [comment]
@myzie this is ready for a firs pass :eyes:.

Similar to `aws` and other optional modules, made it optional too.

File: modules/semver/semver.go [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+	"context"
+	"strings"
+
+	"github.com/blang/semver/v4"
+	"github.com/risor-io/risor/arg"
+	"github.com/risor-io/risor/object"
+)
+
+func Parse(ctx context.Context, args ...object.Object) object.Object {
+	if err := arg.Require("semver.build", 1, args); err != nil {
```
Comment (myzie) [inline modules/semver/semver.go:16]:
```suggestion
	if err := arg.Require("semver.parse", 1, args); err != nil {
```

**Comment by myzie** [comment]
Thanks @rubiojr! This looks good to me so far. I played around with it for a couple minutes and it's working as expected.

I should be able to review a bit more and merge soon.

**Comment by rubiojr** [comment]
There's something going on when I `go install`:

```
> cd cmd/risor
> go install .
package github.com/risor-io/risor/cmd/risor
        imports github.com/risor-io/risor/modules/semver: build constraints exclude all Go files in /home/rubiojr/git/rubiojr/risor/modules/semver
```

I haven't been able to figure out yet why that's complaining about that module only, and not other modules like `aws`, where the build tags are also required. `go install -tags semver .` works as expected though.

**Comment by myzie** [comment]
> There's something going on when I `go install`:
> 
> ```
> > cd cmd/risor
> > go install .
> package github.com/risor-io/risor/cmd/risor
>         imports github.com/risor-io/risor/modules/semver: build constraints exclude all Go files in /home/rubiojr/git/rubiojr/risor/modules/semver
> ```
> 
> I haven't been able to figure out yet why that's complaining about that module only, and not other modules like `aws`, where the build tags are also required. `go install -tags semver .` works as expected though.

Aha yeah this is a tricky one with build tags.

I think you need a file like this in your package:
https://github.com/risor-io/risor/blob/main/modules/aws/stub.go

Note the inverted build tag.

**Comment by rubiojr** [comment]
Dang, TIL! thanks!

**Comment by rubiojr** [comment]
@myzie would you like a GitHub workflow that tries to build the interpreter with and without flags, so we detect breakage like the one I introduced? Happy to take care of that in a new PR if desired.

**Review by myzie — APPROVED** [review]
Thank you!

**Pull request merged**